LabExT/Movement/PathPlanning.py

```diff
@@ -210,14 +210,22 @@ def __init__(
         attractive_gain: tuple
             Attractive gain in the potential field
         """
+        # Stage calibration for this potential field
         self.calibration = calibration
-        self.target_position = target_position
 
+        # Fields settings
         self.grid_size = grid_size
         self.grid_outline = grid_outline
         self.repulsive_gain = repulsive_gain
         self.attractive_gain = attractive_gain
 
+        # Read current stage position
+        with self.calibration.perform_in_system(CoordinateSystem.CHIP):
+            self.start_coordinate = self.calibration.get_position()
```

Comment:
I suggest you add an error check. Since we know at which z-level we start (`self.stage_z_level`), simply check this value to be somewhat close to the target coordinates z-level (`target_coordinate.z`). If it's more than 1um away, raise an error.

Reply:
Okay, but for this I need to pass the z-lift to the PathPlanning. like I had before. 😅

Reply:
Umm, why? You have both variables `self.stage_z_level` and `target_coordinate.z` available in your local scope. No need for any external flag.

Reply:
`target_coordinate.z` is always 0 unless I change the z-value before passing it to the planner.
And `self.stage_z_level` is always going to be `start_coordinate.z` unless I pass more information to the planner. So I cannot really implement the check on the Planner level (without passing more information to it)

Or do I miss something here?

Reply:
Sorry for the delay.

No, you are fine. Still the check makes sense in the planner to make sure that start an end z-heights are equal. Because this is a fundamental assumption in your planner.

It does not matter if this check always passes with the current implementation of mover. Generally, adding checks to see if the inputs to an algorithm adhere to the assumptions that said algorithm works with is always (IMO) a good idea, no?

Reply:
I agree! Opened #114 for this and merge this PR for now 😄